src/main/java/Board.java

```diff
@@ -0,0 +1,117 @@
+import java.util.Arrays;
+
+public class Board {
+
+    private Player[][] grid;
```

Comment:
Consider whether you can find a better way to store the board than a 2-dimensional array. Methods in this class are more complicated than they need to be, and I don't see the advantage of the 2d array. Also, having to pass the row and column to all sorts of methods is a little smelly to me.